pkg/discovery/network/generic.go

```diff
@@ -62,13 +50,112 @@ func discoverGenericNetwork(clientSet kubernetes.Interface) (*ClusterNetwork, er
 }
 
 func findClusterIPRange(clientSet kubernetes.Interface) (string, error) {
+	clusterIPRange, err := findClusterIPRangeApiserver(clientSet)
+	if err == nil && clusterIPRange != "" {
+		return clusterIPRange, nil
+	}
+
+	clusterIPRange, err = findClusterIPRangeSvcCreation(clientSet)
+	if err == nil && clusterIPRange != "" {
+		return clusterIPRange, nil
+	}
+
+	return "", fmt.Errorf("clusterIPRange can't be found by any methods")
+}
+
+func findClusterIPRangeApiserver(clientSet kubernetes.Interface) (string, error) {
 	return findPodCommandParameter(clientSet, "component=kube-apiserver", "--service-cluster-ip-range")
 }
 
+func findClusterIPRangeSvcCreation(clientSet kubernetes.Interface) (string, error) {
+	// find service cidr based on https://stackoverflow.com/questions/44190607/how-do-you-find-the-cluster-service-cidr-of-a-kubernetes-cluster
```

Comment:
It's not possible out of cluster-info dump. That's what our other heuristics do already. But in some cases, the api server services don't run inside the cluster itself (for example they are provided directly by GKE, AKS... or in the case of rancher they run in a container outside of kubernetes, and not inside), so the details can't be found.

I believe that the extra heuristic that mkimura added is great, and will find the range in many cases. If this still fails, it's OK, then the admin will be asked for the range, and life goes on, but this will catch 90% of the remaining cases for AKS,GKE,rancher...

Reply:
I tend to agree with @mangelajo as we have some implementations of K8s which run the API server outside of the cluster and if we really want to purse cluster-info dump, it can be done in a separate PR.